core/src/main/java/de/valtech/aecu/core/service/AecuMigrationServiceImpl.java

```diff
@@ -0,0 +1,89 @@
+package de.valtech.aecu.core.service;
+
+import de.valtech.aecu.api.service.AecuException;
+import de.valtech.aecu.api.service.AecuMigrationService;
+import de.valtech.aecu.api.service.AecuService;
+import de.valtech.aecu.core.serviceuser.ServiceResourceResolverService;
+import de.valtech.aecu.core.util.runtime.RuntimeHelper;
+import java.util.List;
+import javax.jcr.Session;
+import org.apache.sling.api.resource.LoginException;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@link de.valtech.aecu.api.service.AecuMigrationService} impl that executes the AECU migration based on node store type
+ */
+@Component(
+        service = AecuMigrationService.class,
+        immediate = true,
+        name = "AECU migration service"
+)
+public class AecuMigrationServiceImpl implements AecuMigrationService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AecuMigrationServiceImpl.class);
+
+    @Reference
+    private AecuService aecuService;
+    @Reference
+    private ServiceResourceResolverService resourceResolverService;
+
+    @Activate
+    public void activate() {
+        ResourceResolver resourceResolver = getResourceResolver();
+        Session session = resourceResolver.adaptTo(Session.class);
+        boolean isCompositeNodeStore = RuntimeHelper.isCompositeNodeStore(session);
+        if(isCompositeNodeStore) {
+            startAecuMigration(resourceResolver);
+        }
+    }
+
+    /**
+     * Starts the AECU migration
+     * @param resourceResolver the resource resolver to use
+     */
+    void startAecuMigration(ResourceResolver resourceResolver) {
+        try {
+            LOGGER.info("AECU migration started");
+            List<String> migrationScripts = aecuService.getFiles(AecuService.AECU_APPS_PATH_PREFIX);
+            if (!migrationScripts.isEmpty()) {
```

Comment:
Please use AecuService.executeWithInstallHookHistory() with the apps directory as parameter. This way you do not need to list fles and execute one-by-one. It will also generate just one history entry.